test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -261,6 +270,8 @@ private static void validateSpec(RestSpec restSpec) {
 
     @After
     public void wipeCluster() throws Exception {
+        // Running tasks might create indices and otherwise invalidate the wipe. Lets just let the finish.
+        waitForAllRunningTasksToFinish();
```

Comment:
I guess I have 2 concerns. First is a user deleting an index and it causing problems. We shouldn't let this happen; deleting an index should be allowed at any time and it shouldn't cause the user to get into a messed up state. Second is having this custom wait method, when it seems we should just be able to cancel all running tasks, akin to deleting all indices?